### ISSUE TITLE
Proxy crashes when there is an empty response

### ISSUE THREAD

**Issue by jorenham** [wkeeling/selenium-wire#102]
This occurs rarely with my proxy settings but crashes my program. Is there any way to handle this gracefully?

```
Traceback (most recent call last):,
  File "/usr/local/lib/python3.8/socketserver.py", line 650, in process_request_thread,
    self.finish_request(request, client_address),
  File "/usr/local/lib/python3.8/socketserver.py", line 360, in finish_request,
    self.RequestHandlerClass(request, client_address, self),
  File "/usr/local/lib/python3.8/site-packages/seleniumwire/proxy/handler.py", line 164, in __init__,
    super().__init__(*args, **kwargs),
  File "/usr/local/lib/python3.8/site-packages/seleniumwire/proxy/proxy2.py", line 47, in __init__,
    super().__init__(*args, **kwargs),
  File "/usr/local/lib/python3.8/socketserver.py", line 720, in __init__,
    self.handle(),
  File "/usr/local/lib/python3.8/http/server.py", line 426, in handle,
    self.handle_one_request(),
  File "/usr/local/lib/python3.8/http/server.py", line 414, in handle_one_request,
    method(),
  File "/usr/local/lib/python3.8/site-packages/seleniumwire/proxy/proxy2.py", line 75, in do_GET,
    self.admin_handler(),
  File "/usr/local/lib/python3.8/site-packages/seleniumwire/proxy/handler.py", line 30, in admin_handler,
    self._get_requests(),
  File "/usr/local/lib/python3.8/site-packages/seleniumwire/proxy/handler.py", line 67, in _get_requests,
    self._send_response(json.dumps(self.server.storage.load_requests()).encode(,
  File "/usr/local/lib/python3.8/site-packages/seleniumwire/proxy/storage.py", line 166, in load_requests,
    request = self._load_request(indexed_request.id),
  File "/usr/local/lib/python3.8/site-packages/seleniumwire/proxy/storage.py", line 179, in _load_request,
    response = pickle.load(res),
FError: Ran out of input,
```

**Comment by wkeeling** [comment]
It looks as though Selenium Wire has created a zero-byte file when storing a request, and then that triggers the `FError` when it attempts to load that stored request. 

Can you see whether it is a specific request that is triggering the problem, or whether it's just happening randomly? If a specific request, it may be possible to filter out that request by using [request scoping](https://github.com/wkeeling/selenium-wire#scoping-request-capture). That may offer an immediate workaround.

Longer term we'd need to understand why the zero byte file is being created in the first place, and handle that situation better - which may need a code fix.

**Comment by wkeeling** [comment]
Although I'm not entirely sure how a zero-byte response file has been created, a fix has gone in which will handle any `EOFError` and will resolve this particular problem. Available in the next release.

**Comment by wkeeling** [comment]
Fixed in version 1.1.0

**Comment by KeeonTabrizi** [comment]
@wkeeling I'm getting this exception on v1.2.2. Is there a good way on my side to handle these sorts of errors?

```
Exception happened during processing of request from ('127.0.0.1', 59074)
Traceback (most recent call last):
  File "/usr/lib/python3.8/socketserver.py", line 650, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python3.8/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "...venv/lib/python3.8/site-packages/seleniumwire/proxy/handler.py", line 164, in __init__
    super().__init__(*args, **kwargs)
  File "...venv/lib/python3.8/site-packages/seleniumwire/proxy/proxy2.py", line 33, in __init__
    super().__init__(*args, **kwargs)
  File "/usr/lib/python3.8/socketserver.py", line 720, in __init__
    self.handle()
  File "/usr/lib/python3.8/http/server.py", line 427, in handle
    self.handle_one_request()
  File ".../venv/lib/python3.8/site-packages/seleniumwire/proxy/proxy2.py", line 246, in handle_one_request
    super().handle_one_request()
  File "/usr/lib/python3.8/http/server.py", line 415, in handle_one_request
    method()
  File ".../venv/lib/python3.8/site-packages/seleniumwire/proxy/proxy2.py", line 57, in proxy_request
    self.admin_handler()
  File ".../venv/lib/python3.8/site-packages/seleniumwire/proxy/handler.py", line 30, in admin_handler
    self._get_requests()
  File ".../venv/lib/python3.8/site-packages/seleniumwire/proxy/handler.py", line 67, in _get_requests
    self._send_response(json.dumps(self.server.storage.load_requests()).encode(
  File ".../venv/lib/python3.8/site-packages/seleniumwire/proxy/storage.py", line 166, in load_requests
    request = self._load_request(indexed_request.id)
  File ".../venv/lib/python3.8/site-packages/seleniumwire/proxy/storage.py", line 175, in _load_request
    request = pickle.load(req)
EOFError: Ran out of input
----------------------------------------
```

**Comment by wkeeling** [comment]
@KeeonTabrizi interestingly this `EOFError` seems to be happening when a zero-byte _request_ has been stored. I'm not entirely sure why that would happen. 

Do you see this on all requests or just some requests? Does it happen on v1.2.1?

**Comment by KeeonTabrizi** [comment]
@wkeeling - I would see this regularly when I was using v1.0.7. It happens on a small subset of requests and it seems unpredictable. I will downgrade to v1.2.1 to see today and report back if i see it.

---

Tangentially - there are some new errors I've been seeing on 1.2.2/1.2.3 that I did not previously see which I'm trying to make sense of. Have been thinking whether I should make a new issue or try to debug. Mostly these sorts of exceptions that I did not see before.

```
ERROR:seleniumwire.proxy.handler:Error making request
...
socket.timeout: _ssl.c:1091: The handshake operation timed out

or

ERROR:seleniumwire.proxy.handler:Error making request
...
socket.timeout: timed out
```
